ROX-30739: Docs for process baseline auto locking (#100444)
openshift/openshift-docs: modules/auto-lock-process-baselines-known-limitations.adoc

// Module included in the following assemblies:
//
// * operating/evaluate-security-risks.adoc
:_mod-docs-content-type: CONCEPT
[id="auto-lock-process-baselines-known-limitations_{context}"]
= Auto-lock process baselines known limitations

[role="_abstract"]
Central, Central DB, and Sensor consume more CPU and memory resources when process baseline auto-locking is enabled. This can lead to CPU throttling and pods crashing due to running out of memory.

The following results were obtained from tests with 1,000 deployments in which 5,000 process were spawned every 30 seconds (166.67 processes per second). The test was run with the feature enabled and disabled. Resource usage was compared between the two tests. For the tests the process baseline generation duration was set to three minutes and the rate of process creation did not change after the baseline generation period ended.

* Sensor used 24 Mb more memory.

> * The difference in sensor memory usage didn’t seem to be increasing with time.

Reviewer: OK, if we are leaving these in here, there are a couple of changes I would suggest.

```suggestion
* The difference in sensor memory usage apparently did not appear to increase over time.
```

* Sensor CPU usage increase by 0.14 CPUs.
* Central used 175 Mb more memory.
* The rate of increase of central memory usage was 65 Kb per second greater.
* Central CPU usage increase by 0.12 CPUs.
* Central DB used 296 Mb more memory.

> * The difference in central-db memory usage didn’t seem to be increasing with time.

Reviewer: ```suggestion
* The difference in central-db memory usage did not appear to increase over time.
```
